src/plone/app/blob/subtypes/image.py

```diff
@@ -21,7 +21,11 @@ class ExtensionBlobField(ExtensionField, BlobField, ImageFieldMixin):
     def set(self, instance, value, refresh_exif=True, **kwargs):
         super(ExtensionBlobField, self).set(instance, value, **kwargs)
         self.fixAutoId(instance)
-        instance.getEXIF(value, refresh=refresh_exif)
+        if hasattr(instance, 'getEXIF'):
+            # If the instance subclasses ATCTImageTransform we process the
+            # metadata.
+            # TODO: The EXIF functionality should not be dependent on ATCT
```

Comment:
I suspect it should be a method on the field rather than on the content instance.

Reply:
It absolutely should, but the EXIF functionality is quite extensive. It would be pretty horrible to have to duplicate it and it appears to be written with the assumption that the class is co-operating. Perhaps it could be refactored as an adapter or utility functions, but it seemed to me that as the EXIF functionality is progressive enhancement that this was a good compromise.

Reply:
👍